crates/phactory/src/contracts/pink.rs

```diff
@@ -503,12 +514,23 @@ impl Cluster {
                 transfer,
                 estimating,
             } => {
+                let origin = origin.cloned().ok_or(QueryError::BadOrigin)?;
```

Comment:
Is it a problem before without this verification?

Reply:
There is not much difference in safety. This line of code was at a few lines below, and I moved it up. Filtering out None origin before requesting the scheduler looks slightly better, but for the requester, filling in the origin does not require any cost.